src/fstapp_mnesia_handler.erl

```diff
@@ -18,11 +18,11 @@
 handle_init() ->
 	_ =  mnesia:create_schema([node()]),
 	_ = mnesia:create_table(metrics,[{attributes, record_info(fields, metrics)}]),
-	ok.
+	{ok, ok}.
 
-handle_data(Metrics) ->
+handle_data(Metrics, InternalState) ->
```

Comment:
There is no need to call it `InternalState`.  It is InternalState in the gen_server context but here can be called just __State__